src/pattern-library/components/patterns/prototype/EnhancedSelectPage.tsx

```diff
@@ -0,0 +1,194 @@
+  );
+  const next = useCallback(() => {
+    const newIndex = selectedIndex + 1;
+    setSelected(
+      (items[newIndex]?.id === 4 ? items[newIndex + 1] : items[newIndex]) ??
+        selected,
+    );
+  }, [selected, selectedIndex]);
+  const previous = useCallback(() => {
+    const newIndex = selectedIndex - 1;
+    setSelected(
+      (items[newIndex]?.id === 4 ? items[newIndex - 1] : items[newIndex]) ??
+        selected,
+    );
+  }, [selected, selectedIndex]);
+
+  return (
+    <InputGroup>
+      <IconButton
+        icon={ArrowLeftIcon}
+        title="Previous student"
+        variant="dark"
+        onClick={previous}
+        disabled={selectedIndex <= 0}
+      />
+      <div className="w-[350px]">
+        <Select
+          selected={selected}
+          onChange={setSelected}
+          label={
+            selected ? (
+              <>
+                {selected.name}
+                <div className="rounded px-2 bg-grey-7 text-white">
+                  {selected.id}
+                </div>
+              </>
+            ) : (
+              <>Select one...</>
+            )
+          }
+        >
+          {items.map(item => (
+            <Select.Option value={item} key={item.id} disabled={item.id === 4}>
+              {({ disabled }) => (
+                <>
+                  <span
+                    className={
+                      disabled ? 'line-through text-grey-6' : undefined
+                    }
+                  >
+                    {item.name}
+                  </span>
+                  <div className="grow" />
+                  <div
+                    className={classnames('rounded px-2 text-white', {
+                      'bg-grey-7': !disabled,
+                      'bg-grey-4': disabled,
+                    })}
+                  >
+                    {item.id}
+                  </div>
+                </>
+              )}
+            </Select.Option>
+          ))}
+        </Select>
+      </div>
+      <IconButton
+        icon={ArrowRightIcon}
+        title="Next student"
+        variant="dark"
+        onClick={next}
+        disabled={selectedIndex >= items.length - 1}
+      />
+    </InputGroup>
+  );
+}
+
+export default function EnhancedSelectPage() {
+  return (
+    <Library.Page
+      title="Enhanced Select"
+      intro={
+        <p>
+          Experimentation to get an enhanced <code>Select</code> where options
+          can be styled.
+        </p>
+      }
+    >
+      <Library.Section>
+        <Library.Pattern title="Select">
+          <Select_ />
+        </Library.Pattern>
+
+        <Library.Pattern title="Select with many options">
+          <Select_
+            theItems={[
+              ...items,
+              ...items,
+              ...items,
+              ...items,
+              ...items,
+              ...items,
+            ]}
+          />
+        </Library.Pattern>
+
+        <Library.Pattern title="Plain text Select">
```

Comment:
I would call the plain text select just "Select" and the example with badges "Select (custom items)" or similar.